src/main/java/seedu/student/storage/JsonSerializableStudentBook.java

```diff
@@ -48,22 +50,35 @@ public JsonSerializableStudentBook(ReadOnlyStudentBook source) {
     }
 
     /**
-     * Converts this student book into the model's {@code AddressBook} object.
+     * Converts this student book into the model's {@code StudentBook} object.
      *
      * @throws IllegalValueException if there were any data constraints violated.
      */
     public StudentBook toModelType() throws IllegalValueException {
         StudentBook studentBook = new StudentBook();
         for (JsonAdaptedStudent jsonAdaptedStudent : students) {
             Student student = jsonAdaptedStudent.toModelType();
+
             if (studentBook.hasStudent(student)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_STUDENT);
             }
             studentBook.addStudent(student);
         }
         for (JsonAdaptedAppointment jsonAdaptedAppointment : appointments) {
             Appointment appointment = jsonAdaptedAppointment.toModelType();
-            if (studentBook.hasAppointment(appointment)) {
+
+            Boolean isExistingStudent = studentBook.getStudentList().stream()
+                    .anyMatch(student -> student.getMatriculationNumber().equals(appointment.getMatriculationNumber()));
```

Comment:
Should you use the method `StudentBook::isExistingMatricNumber` that yien implemented for this instead? Also do we need reference-type `Boolean` or is `boolean` sufficient?

Reply:
We should use the <StudentBook::isExistingMatricNumber> method and <boolean> is sufficient.  Thanks for pointing this out!